src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js

```diff
@@ -68,30 +68,34 @@ const DomainList = (props) => {
         </TouchableNativeFeedback>
       );
     }
-    if (item.isunfollowed) {
+    if (item.withFollowButton) {
+      if (item.isunfollowed) {
+        return (
+          <TouchableNativeFeedback onPress={handleSetFollow}>
+            <View style={styles.followContainer}>
+              <View style={styles.buttonFollow}>
+                <Text style={styles.textButtonFollow}>{isCommunity ? 'Join' : 'Follow'}</Text>
+              </View>
+            </View>
+          </TouchableNativeFeedback>
+        );
+      }
       return (
-        <TouchableNativeFeedback onPress={handleSetFollow}>
+        <TouchableNativeFeedback onPress={handleSetUnFollow}>
           <View style={styles.followContainer}>
-            <View style={styles.buttonFollow}>
-              <Text style={styles.textButtonFollow}>{isCommunity ? 'Join' : 'Follow'}</Text>
+            <View style={styles.buttonFollowing}>
+              {isFromUserFragment ? (
+                <MemoIc_senddm height={20} width={20} color={COLORS.signed_primary} />
+              ) : (
+                <Text style={styles.textButtonFollowing}>
+                  {isCommunity ? 'Joined' : 'Following'}
+                </Text>
+              )}
             </View>
           </View>
         </TouchableNativeFeedback>
       );
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new condition `item.withFollowButton` is a good addition to control the rendering of follow/unfollow buttons. However, it's not clear where `handleSetFollow` and `handleSetUnFollow` functions are defined and what they do. Make sure these functions handle potential errors and edge cases properly. Also, consider renaming `isunfollowed` to `isUnfollowed` for better readability and consistency with JavaScript naming conventions.
```diff
-    if (item.isunfollowed) {
+    if (item.isUnfollowed) {
```
Also, ensure that the `isFromUserFragment` flag is being set correctly wherever this component is used. This flag changes the UI significantly and could lead to confusion if not managed properly.


<!-- This is an auto-generated comment by OSS CodeRabbit -->